src/scripts/components/stories/pilot-story/pilot-story.tsx

```diff
@@ -30,59 +30,68 @@ import {ChapterContextProvider} from './provider/chapter-provider';
 
 const PilotStory: FunctionComponent = () => {
   const [storyStarted, setStoryStarted] = useState(false);
-
   const {isDesktop} = useScreenSize();
+  const [scrollEl, setScrollElement] = useState<HTMLDivElement | null>(null);
+  const ref = useRef<HTMLDivElement | null>(null);
 
-  return (
-    <ParallaxProvider>
-      <ChapterContextProvider>
-        <div className={styles.pilotStory}>
-          <Header>
-            <Share />
-          </Header>
-          {isDesktop && storyStarted && (
-            <ChapterProgressIndication
-              chapters={chapters}
-              className={styles.chapterProgressIndication}
-              gap={48}
-            />
-          )}
+  useEffect(() => {
```

Comment:
did you try the `useLayoutEffect` hook? I might be fit here. Also, please add a comment to explain what's going on :)